src/BootKeyboard/BootKeyboard.cpp

```diff
@@ -111,6 +117,12 @@ int BootKeyboard_::getDescriptor(USBSetup& setup) {
 
 
 void BootKeyboard_::begin() {
+// We plug into PluggableUSB lazily on AVR to give the compiler a chance to 
+// optimize out the implementation
+#ifdef __AVR__
+  PluggableUSB().plug(this);
+#endif
+
```

Comment:
I believe it's safe to just do it in the constructor on AVR too. That does prevent the compiler from optimizing things out, yes, but one would need to do a few more things to let that happen, and I'm not aware of anyone (apart from myself) who went there.

So to simplify the code, I'd just plug it in the constructor. If we do get a report that it causes size issues on AVR, we can always add this ifdef pair back. But at this time, it feels like unnecessary optimization.

As in, to let the compiler optimize BootKeyboard out, you'd have to create a derived device, with a HID driver that overrides the default `driver::hid::base::Keyboard` methods that reference the boot keyboard. Or one that stubs out BootKeyboard with a dummy.

The late initialization has not served its purpose since the move to drivers, really.